### ISSUE TITLE
kafka-panache-reactive-quickstart fails after #27802 

### ISSUE THREAD

**Issue by rsvoboda** [quarkusio/quarkus#27873]
### Describe the bug

kafka-panache-reactive-quickstart fails after https://github.com/quarkusio/quarkus/pull/27802 got merged, CC @ozangunalp

https://github.com/quarkusio/quarkus/pull/27802 is breaking change and there is no information in https://github.com/quarkusio/quarkus/wiki/Migration-Guide-2.13

Proposal:
 - update Migration-Guide-2.13
 - indicate which class could use `@ActivateRequestContext` to fix the trouble
(Assuming that we really need @ActivateRequestContext)

Error I receive is
```
2022-09-12 15:39:32,734 ERROR [io.sma.rea.mes.provider] (vert.x-eventloop-thread-4) SRMSG00201: Error caught while processing a message: javax.enterprise.context.ContextNotActiveException: RequestScoped context was not active when trying to obtain a bean instance for a client proxy of PRODUCER_METHOD bean [class=io.quarkus.hibernate.reactive.runtime.ReactiveSessionProducer, id=37346551fd6c9e87c80f25842d68c2d902107673]
	- you can activate the request context for a specific method using the @ActivateRequestContext interceptor binding
	at io.quarkus.arc.impl.ClientProxies.getDelegate(ClientProxies.java:53)
```

I can get the test passing when I adjust https://github.com/quarkusio/quarkus-quickstarts/blob/main/kafka-panache-reactive-quickstart/src/main/java/org/acme/panache/PriceStorage.java#L15 but the QS is small so it was easy to find the right class and method to annotate

```diff
--- a/kafka-panache-reactive-quickstart/src/main/java/org/acme/panache/PriceStorage.java
+++ b/kafka-panache-reactive-quickstart/src/main/java/org/acme/panache/PriceStorage.java
@@ -6,12 +6,14 @@ import io.smallrye.reactive.messaging.annotations.Blocking;
 import org.eclipse.microprofile.reactive.messaging.Incoming;

 import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.control.ActivateRequestContext;
 import javax.transaction.Transactional;

 @ApplicationScoped
 public class PriceStorage {

     @Incoming("prices")
+    @ActivateRequestContext
     Uni<Void> store(int priceInUsd) {
         Price price = new Price();
         price.value = priceInUsd;
```

### Expected behavior

Tests pass

### Actual behavior

Tests fail

### How to Reproduce?

 - clone https://github.com/quarkusio/quarkus-quickstarts
 - run `mvn clean verify -pl kafka-panache-reactive-quickstart`

### Output of `uname -a` or `ver`

macOS Monterey

### Output of `java -version`

Java 17

### Quarkus version or git rev

Quarkus main

**Comment by cescoffier** [comment]
It's actually a bug fix. It was not really working... it was using the request scope from the application startup.

**Comment by rsvoboda** [comment]
I'm not proposing to revert it;)

I want to have it documented in migration guide as it's a breaking change

And if possible to indicate which class could use @ActivateRequestContext so the end users can fix their code a bit faster

**Comment by rsvoboda** [comment]
(I will proceed with QS fix in the meantime)

**Comment by cescoffier** [comment]
I've asked Ozan to add a note (it should have been done while merging)

**Comment by ozangunalp** [comment]
I've missed that. 
@rsvoboda thanks for the heads up, I'll update the migration guide accordingly. 

Too bad the error message doesn't mention the channel name. I'll also look into it.

**Comment by ozangunalp** [comment]
I think we can close this
- Quickstart fixed https://github.com/quarkusio/quarkus-quickstarts/pull/1174 
- 2.13 Migration guide updated : https://github.com/quarkusio/quarkus/wiki/Migration-Guide-2.13
- Descriptive logging fixed : https://github.com/smallrye/smallrye-reactive-messaging/pull/1889

**Comment by rsvoboda** [comment]
When is `3.20.0` planned for release? And will it make it into Quarkus 2.13 branch?

**Comment by ozangunalp** [comment]
I'll try to open the version bump PR this week for 3.19.0 but it doesn't include the last item.
3.20.0 should follow shortly. But I don't know which Quarkus milestone it'll correspond.

**Comment by cescoffier** [comment]
That issue should be closed.
3.20.0 log fix is nice to have, but not something required.
